textproto/mod.ts

```diff
@@ -83,7 +83,13 @@ export class TextProtoReader {
 
     while (true) {
       let [kv, err] = await this.readLineSlice(); // readContinuedLineSlice
-      if (kv.byteLength == 0) {
+
+      // If we encounter an empty line with a space
+      // it's the end of mime headers
+      if (
+        kv.length === kv.filter((x): boolean => x === charCode(" ")).length ||
+        kv.byteLength === 0
+      ) {
```

Comment:
This `while (true)` loop corresponds exactly to this code:
https://github.com/golang/go/blob/1ab063ce532f72851cef735238ba656cc7680b66/src/net/textproto/reader.go#L493-L540

Rather than doing this kv.filter thing, can you try to match the logic exactly as they have it?

Reply:
I think readlineSlice miss some part of the golang function:
https://github.com/golang/go/blob/1ab063ce532f72851cef735238ba656cc7680b66/src/net/textproto/reader.go#L123

This part:
https://github.com/golang/go/blob/1ab063ce532f72851cef735238ba656cc7680b66/src/net/textproto/reader.go#L150

I'm change this is better this way yes.